normalized-cache-incubating/src/commonMain/kotlin/com/apollographql/cache/normalized/CacheMissLoggingInterceptor.kt

```diff
@@ -15,7 +15,7 @@ class CacheMissLoggingInterceptor(private val log: (String) -> Unit) : ApolloInt
       if (it.exception is CacheMissException) {
         log(it.exception!!.message.toString())
       } else
-        it.errors.orEmpty().mapNotNull { it.extensions?.get("exception") as? CacheMissException }.forEach {
+        it.errors.orEmpty().mapNotNull { it.exception as? CacheMissException }.forEach {
```

Comment:
Storing exceptions in `extensions` feels conceptually wrong. `extensions` is in the JSON domain. Technically, it should probably be a `Map<String, ApolloJsonElement>`.

Reply:
I agree...

To rewind a bit: I want to surface structured info about the cache miss errors, and `CacheMissException` is a good fit.

So I could store that info as `ApolloJsonElement` and convert it back to a `CacheMissException` when needed but that's a bit of overhead for no great reason other than "purity".

Another thing we could do is add an `apolloExtensions` field on `Error` to make it clear that this can contain extensions from the library, not from the server.

Reply:
Purity is a great reason IMO 😄 I like purity!

That aside, looks like that information is always used in an aggregate way (either for logging or added as suppressed exceptions). You _could_ have `BatchCacheReader` return side info about the cache misses found along the way.